app/src/lib/text-token-parser.ts

```diff
@@ -338,7 +336,7 @@ export class Tokenizer {
    *
    * @returns an array of tokens representing the scan results.
    */
-  public tokenize(text: string): ReadonlyArray<TokenResult> {
+  public tokenize(text: string): TokenResult[] {
```

Comment:
Since this is a public method and it's used from React components I'm not super happy about making the output array mutable (with these current changes it's still safe but I'm worried that at some point somebody will pass this mutable around as a Prop to a React component).

Can we keep this as a readonly array? (see my comment below)

Reply:
> at some point somebody will pass this mutable around as a Prop to a React component

You sure know how to nerd-snipe me. See https://github.com/desktop/desktop/pull/9939

> Can we keep this as a readonly array? (see my comment below)

Yep, I took a stab at reworking it, I took a slightly different approach to yours but I think we achieve the same end-result.